test/services/samples/metadata/file_import_service_test.rb

```diff
@@ -0,0 +1,159 @@
+        )
+        assert_equal({ 'metadatafield1' => 10, 'metadatafield2' => 20, 'metadatafield3' => 30 },
+                     @sample1.reload.metadata)
+        assert_equal({ 'metadatafield1' => 15, 'metadatafield2' => 25, 'metadatafield3' => 35 },
+                     @sample2.reload.metadata)
+      end
+
+      test 'import sample metadata via xlsx file' do
+        xlsx = File.new('test/fixtures/files/metadata/valid.xlsx', 'r')
+        params = { file: xlsx, sample_id_column: 'sample_name' }
+        response = Samples::Metadata::FileImportService.new(@project, @john_doe,
+                                                            params).execute
+        assert_equal(
+          { @sample1.name => { updated: %w[metadatafield1 metadatafield2 metadatafield3], not_updated: [] },
+            @sample2.name => { updated: %w[metadatafield1 metadatafield2 metadatafield3], not_updated: [] } }, response
+        )
+        assert_equal({ 'metadatafield1' => 10, 'metadatafield2' => 20, 'metadatafield3' => 30 },
+                     @sample1.reload.metadata)
+        assert_equal({ 'metadatafield1' => 15, 'metadatafield2' => 25, 'metadatafield3' => 35 },
+                     @sample2.reload.metadata)
+      end
+
+      test 'import sample metadata via other file' do
+        other = File.new('test/fixtures/files/metadata/invalid.txt', 'r')
+        params = { file: other, sample_id_column: 'sample_name' }
+        assert_empty Samples::Metadata::FileImportService.new(@project, @john_doe,
+                                                              params).execute
+      end
+
+      test 'import sample metadata with no sample_id_column' do
+        csv = File.new('test/fixtures/files/metadata/missing_sample_id_column.csv', 'r')
+        params = { file: csv, sample_id_column: 'sample_name' }
+        assert_empty Samples::Metadata::FileImportService.new(@project, @john_doe,
+                                                              params).execute
+      end
+
+      test 'import sample metadata with no metadata columns' do
+        csv = File.new('test/fixtures/files/metadata/missing_metadata_columns.csv', 'r')
+        params = { file: csv, sample_id_column: 'sample_name' }
+        assert_empty Samples::Metadata::FileImportService.new(@project, @john_doe,
+                                                              params).execute
+      end
+
+      test 'import sample metadata with no metadata rows' do
+        csv = File.new('test/fixtures/files/metadata/missing_metadata_rows.csv', 'r')
+        params = { file: csv, sample_id_column: 'sample_name' }
+        assert_empty Samples::Metadata::FileImportService.new(@project, @john_doe,
+                                                              params).execute
+      end
+
+      test 'import sample metadata with empty values set to true' do
+        @sample1.metadata = { 'metadatafield1' => '1', 'metadatafield2' => '2', 'metadatafield3' => '3' }
```

Comment:
Eric mentioned in a PR of mine he wanted me to get away from setting the metadata like so in a test and to use fixtures with metadata already added.
[PR339](https://github.com/phac-nml/irida-next/pull/339) also contains a bunch of new fixtures that contains metadata.

@ericenns did you want @ksierks to utilize these fixtures once merged into `main`?

Reply:
Please see 8dab7b7fb656212806f88f0332ae792c31a1f2e1.